### DISCUSSION TITLE
Pass list as `rowname_col` parameter

### DISCUSSION THREAD

**Discussion by abdullahau** [posit-dev/great-tables#512] [Q&A]
Hi,

I was trying to create the correlation matrix from a Polars dataframe. Unlike Pandas, the .corr() method does not create an index where the column names appear on each row of the matrix. Is there a way to pass lists instead of column references for rowname? I could see this being really helpful while using Polars

Here is my workaround:
```python
# Compute the Correlation Matrix
corr_matrix = (
    pct_volume.select(
        pl.col('*').exclude('week')  # Exclude 'week' column if applicable
    )
    .corr()  # Compute the correlation matrix
    .hstack(pl.DataFrame({"index": pct_volume.columns[1:]}))
)

(
    GT(corr_matrix, rowname_col='index')
    .tab_header(title="Correlation Matrix of Weekly (Volume) Market Share")
    .fmt_number(columns=['Alpha', 'Bravo', 'Charlie', 'Delta', 'Other'])
    .data_color(
        domain=[-1, 1],
        palette=["rebeccapurple", "white", 'orange']
    )
)
```

If you have suggestions for improvement, please let me know.

**Comment by jrycw** [comment]
@abdullahau , thanks for bringing up this discussion. Personally, I believe it’s possible to make some modifications to [`_prep_gt()`](https://github.com/posit-dev/great-tables/blob/main/great_tables/_gt_data.py#L94) to achieve this. For now, I’d likely approach this by adding a new column using `df.with_columns()` like:  
```python
import polars as pl
from great_tables import GT


df = pl.DataFrame({"foo": [1, 2, 3], "bar": [3, 2, 1], "ham": [7, 8, 9]})
corr_matrix = df.corr().with_columns(pl.Series(df.columns).alias("index"))
(GT(corr_matrix, rowname_col="index"))
```

**Reply by abdullahau** [reply]
Thank you for the suggestion, @jrycw! This method aligns so well with Polars' syntax.

**Comment by jrycw** [comment]
I'll close the discussion since the issue appears to be resolved.
